odoo/orm/fields_misc.py

```diff
@@ -93,6 +93,7 @@ def convert_to_export(self, value, record):
 
 class Id(Field[IdType | typing.Literal[False]]):
     """ Special case for field 'id'. """
+    # Note: This field type is not necessarily an integer!
     type = 'integer'  # note this conflicts with Integer
```

Comment:
I am wondering about what could possibly go wrong with this, in case `id` is actually a string.
- Having the `id` field in a view might crash, since the client would use the `integer` widget by default.
- Some generic business code relying on `field.type` might crash as well.  In most cases, the ORM considers `id` as a special case.  We should never rely on `field.type` for field `id` in the ORM!
- Could CSV exports be problematic?  I don't know.  Imports shouldn't be a problem, since the model should not be editable.

Reply:
Actually, I find it strange that `type = 'integer'`, I think it should be "id". I will check the impact for this change on master.

All relationships (many2one, references, etc.) expect an integer. They just won't work. It could be nice to have a uuid as the identifier, but I don't think it is a priority right now and have a big impact.
Imports have a try-catch to handle something else than int that is passed as-is to the search.